eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/ResourceProvider.cs

```diff
@@ -0,0 +1,358 @@
+            }
+            return tryStatement;
+        }
+
+        private static CSharpType GetResponseType(MethodProvider convenienceMethod, bool isAsync) => isAsync ? convenienceMethod.Signature.ReturnType?.Arguments[0]! : convenienceMethod.Signature.ReturnType!;
+
+        private ValueExpression[] PopulateArguments(IReadOnlyList<ParameterProvider> parameters, MethodProvider convenienceMethod)
+        {
+            var arguments = new List<ValueExpression>();
+            foreach (var parameter in parameters)
+            {
+                if (parameter.Name.Equals("subscriptionId", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    arguments.Add(Static(typeof(Guid)).Invoke(nameof(Guid.Parse), This.Property(nameof(ArmResource.Id)).Property(nameof(ResourceIdentifier.SubscriptionId))));
+                }
+                else if (parameter.Name.Equals("resourceGroupName", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    arguments.Add(This.Property(nameof(ArmResource.Id)).Property(nameof(ResourceIdentifier.ResourceGroupName)));
+                }
+                // TODO: handle parents
+                else if (parameter.Name.Equals(_contextualParameters.Last(), StringComparison.InvariantCultureIgnoreCase))
+                {
+                    arguments.Add(This.Property(nameof(ArmResource.Id)).Property(nameof(ResourceIdentifier.Name)));
+                }
+                else if (parameter.Type.Equals(typeof(RequestContent)))
+                {
+                    var resource = convenienceMethod.Signature.Parameters.Single(p => p.Type.Equals(_resourceData.Type));
+                    arguments.Add(resource);
+                }
+                else if (parameter.Type.Equals(typeof(RequestContext)))
+                {
+                    var cancellationToken = convenienceMethod.Signature.Parameters.Single(p => p.Type.Equals(typeof(CancellationToken)));
+                    arguments.Add(New.Instance(typeof(RequestContext), new Dictionary<ValueExpression, ValueExpression> { { Identifier(nameof(RequestContext.CancellationToken)), cancellationToken } }));
+                }
+                else
+                {
+                    arguments.Add(parameter);
+                }
+            }
+            return arguments.ToArray();
+        }
+
+        // TODO: get clean name of operation Name
+        private MethodProvider GetCorrespondingConvenienceMethod(InputOperation operation, bool isAsync)
+        {
+            // TODO: use _clientProvider.CanonicalView instead when it implements BuildMethods
+            MethodProvider[] methods = [.. _clientProvider.Methods, .. _clientProvider.CustomCodeView?.Methods ?? []];
+            return methods.Single(m => m.Signature.Name.Equals(isAsync ? $"{operation.Name}Async" : operation.Name, StringComparison.OrdinalIgnoreCase) && m.Signature.Parameters.Any(p => p.Type.Equals(typeof(CancellationToken))));
```

Comment:
It should be `operation.Name.ToCleanName()` instead of `operation.Name`. We need to expose [StringExtension.ToCleanName](https://github.com/microsoft/typespec/blob/59f085dbebd12d9f35e4f54926347663eb4bd934/packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator/src/Shared/StringExtensions.cs#L19)